webapp/components/admin_console/login_settings.jsx

```diff
@@ -20,23 +21,87 @@ export default class LoginSettings extends AdminSettings {
 
         this.renderSettings = this.renderSettings.bind(this);
 
+        this.getSampleErrorMsg = this.getSampleErrorMsg.bind(this);
+
         this.state = Object.assign(this.state, {
+            passwordMinimumLength: props.config.PasswordSettings.MinimumLength,
+            passwordLowercase: props.config.PasswordSettings.Lowercase,
+            passwordNumber: props.config.PasswordSettings.Number,
+            passwordUppercase: props.config.PasswordSettings.Uppercase,
+            passwordSymbol: props.config.PasswordSettings.Symbol,
             passwordResetSalt: props.config.EmailSettings.PasswordResetSalt,
             maximumLoginAttempts: props.config.ServiceSettings.MaximumLoginAttempts,
             enableMultifactorAuthentication: props.config.ServiceSettings.EnableMultifactorAuthentication
         });
+
+        // Update sample message from config settings
+        let sampleErrorMsgId = 'user.settings.security.passwordError';
+        if (props.config.PasswordSettings.Lowercase) {
+            sampleErrorMsgId = sampleErrorMsgId + 'Lowercase';
+        }
+        if (props.config.PasswordSettings.Uppercase) {
+            sampleErrorMsgId = sampleErrorMsgId + 'Uppercase';
+        }
+        if (props.config.PasswordSettings.Number) {
+            sampleErrorMsgId = sampleErrorMsgId + 'Number';
+        }
+        if (props.config.PasswordSettings.Symbol) {
+            sampleErrorMsgId = sampleErrorMsgId + 'Symbol';
+        }
+        this.sampleErrorMsg = (
+            <FormattedMessage
+                id={sampleErrorMsgId}
+                default='Your password must be at least {min} characters.'
+                values={{
+                    min: props.config.PasswordSettings.MinimumLength
+                }}
+            />
+        );
+    }
+
+    componentWillUpdate() {
+        this.getSampleErrorMsg();
     }
 
     getConfigFromState(config) {
+        config.PasswordSettings.MinimumLength = parseInt(this.state.passwordMinimumLength, 10);
+        config.PasswordSettings.Lowercase = this.refs.lowercase.checked;
+        config.PasswordSettings.Uppercase = this.refs.uppercase.checked;
+        config.PasswordSettings.Number = this.refs.number.checked;
+        config.PasswordSettings.Symbol = this.refs.symbol.checked;
         config.EmailSettings.PasswordResetSalt = this.state.passwordResetSalt;
         config.ServiceSettings.MaximumLoginAttempts = this.parseIntNonZero(this.state.maximumLoginAttempts);
         if (global.window.mm_license.IsLicensed === 'true' && global.window.mm_license.MFA === 'true') {
             config.ServiceSettings.EnableMultifactorAuthentication = this.state.enableMultifactorAuthentication;
         }
-
         return config;
     }
 
+    getSampleErrorMsg() {
```

Comment:
Since the function is called `getSampleErrorMsg`, I'd expect it to return the error message rather than set a variable. I think this should just return the `FormattedMessage` and you can call it directly in render. It will be ran on every render, but that's not a big deal since it's not particularly complex